### DISCUSSION TITLE
Wave invalid response: Container image does not exist or access is not authorized

### DISCUSSION THREAD

**Discussion by nick-youngblut** [nextflow-io/nextflow#5629] [Q&A]
## Bug report 

If `wave.enabled = false`, my custom pipeline that uses Docker images on the GCP Artifact Registry works correctly (running with the GCP Batch executor).

However, if `wave.enabled = true`, the error is:

```
ERROR ~ Error executing process > '###'

Caused by:
  Wave invalid response: POST https://wave.seqera.io/v1alpha2/container [400] {"message":"Container image 'us-east1-docker.pkg.dev/###/###/###:0.1.0' does not exist or access is not authorized"}
```

So, clearly, the image exists and can be accessed, just not via Wave. There is no clear error message stating why Wave cannot access the image.

### Expected behavior and actual behavior

Wave working with GCP Batch executor and GCP Artifact Registry -- or docs stating that Wave is not supported with the GCP Artifact Registry. 

### Steps to reproduce the problem

GCP Batch + GCP Artifact Registry + Wave

### Program output 

See above

### Environment 

* Nextflow version: 24.10.2
* Java version: 21.0.0
* Operating system: Ubuntu 22.04
* Bash version: 5.1.16

**Comment by pditommaso** [comment]
To access a private repository you need to provide the registry credentials via Platform credentials management. See the docs [here](https://docs.seqera.io/platform/24.1/credentials/google_registry_credentials)

**Comment by nick-youngblut** [comment]
So, a user can just use a private repo via running the pipeline with Seqera Platform?

If so, it would be helpful to include such information in the error message. I'm sure introspection can be used to determine whether the pipeline is being run via the platform, so the message could state that the feature is only available via Seqera Platform instead of `does not exist or access is not authorized`, which is a bit misleading.
